assets/js/components/dashboard-sharing/ModuleRecoveryAlert/index.js

```diff
@@ -128,10 +139,30 @@ export default function ModuleRecoveryAlert() {
 		await recoverModules( modulesToRecover );
 
 		setRecoveringModules( false );
-		setCheckboxes( null );
 
-		return { dismissOnCTAClick: false };
-	}, [ checkboxes, clearRecoveredModules, recoverModules ] );
+		// Dismiss if all recoverable modules are checked or only one module is available for recovery.
+		if (
+			( checkboxes !== null &&
+				recoverableModulesList.every(
+					( module ) => checkboxes[ module ]
+				) ) ||
+			recoverableModulesList.length === 1
```

Comment:
Should we not be using `modulesToRecover` here as the list that was actually used in the action? This and `checkboxes` should be all that's needed I'd think.

Reply:
The aim it so allow the user to select only some of the modules to recover, and the alert continues to be displayed, showing only the remaining modules. I have much simplified this check.